src/uwtools/drivers/driver.py

```diff
@@ -81,13 +81,14 @@ def run_cmd(self) -> str:
         """
         The command-line command to run the NWP tool.
 
-        :return: Collated string that contains MPI command, run time arguments, and exec name.
-        """
-        run_cmd = self._platform_config["mpicmd"]
-        exec_name = self._config["exec_name"]
-        run_time_args = self._config["runtime_info"].get("mpi_args", [])
-        args_str = " ".join(str(arg) for arg in run_time_args)
-        return f"{run_cmd} {args_str} {exec_name}"
+        :return: Collated string that contains MPI command, runtime arguments, and exec name.
+        """
+        components = [
+            self._platform_config.get("mpicmd"),  # MPI run program
+            *[str(x) for x in self._config["runtime_info"].get("mpi_args", [])],  # MPI arguments
+            self._config["exec_name"],  # NWP tool executable name
+        ]
+        return " ".join(filter(None, components))
```

Comment:
See [previous discussion](https://github.com/ufs-community/workflow-tools/pull/327/files#r1380616545) for rationale for this change.